plugins/elantp/fu-elantp-hid-device.c

```diff
@@ -143,6 +144,17 @@ fu_elantp_hid_device_setup (FuDevice *device, GError **error)
 	g_autofree gchar *version_bl = NULL;
 	g_autofree gchar *version = NULL;
 
+	/* get pattern */
+	if (!fu_elantp_hid_device_read_cmd (self,
+					    ETP_CMD_I2C_GET_HID_ID,
+					    buf, sizeof(buf), error)) {
+		g_prefix_error (error, "failed to read HID ID: ");
+		return FALSE;
+	}
+	tmp = fu_common_read_uint16 (buf, G_LITTLE_ENDIAN);
+	if (tmp != 0xffff)
+		self->pattern = (tmp & 0xff00) >> 8;
+
```

Comment:
if(tmp==0xffff) then -> self->pattern = 0;

Reply:
0 is the default, but i'll add it to _init() to make it explicit.